changelog.d/456.fixed.md

```diff
@@ -0,0 +1,2 @@
+Move exclusions up before reading file/dir in _walk_relative_paths.
+erform exclusion checks before file access attempts. This will prevent unnecessary warnings and IO operations on paths that are not relevant to the operation.
```

Comment:
as I user I don't care about code detail, but API behavior changes. Please do not expose internal method changes here.
e.g.
```suggestion
Move scan filters before a read on filesystem access attempt. This will prevent unnecessary warnings and IO operations on paths that are not relevant to the operation.
```

"the unnecessary IO operation" seems kinda untrue, I think at this time with proposed change as is we make more IO operations right now (which is probably negligible, since its only file node metadata hits).

Reply:
Got it. Gonna write with the user in mind